### ISSUE TITLE
menus of main window are obscured by viewports higher in the z-order

### ISSUE THREAD

**Issue by dgm3333** [ocornut/imgui#6453]
Version: 1.89.5
Branch: docking
Back-ends: multiple - this happens to be win32 dx11 - but also occurs with sdl2/ogl3 and glfw
Compiler: Visual Studio
Operating System: Windows 10

The image says it all - if the menu is clicked of a child of the main parent window (aka PCW) then the menu will open *behind* any viewports which happen to be on top of PCW. This doesn't happen if the menu-opening-window started behind but is itself a viewport as in that case it is immediately brought to top when clicked.
I wouldn't have expected that the PCW was brought in front (so that is correct), but I *would* have expected the menu to be drawn on top (accepting this is likely complex as presumably the menu would have to be opened as a new viewport so it doesn't inherit the z-order of the PCW
![image](https://github.com/ocornut/imgui/assets/51210/ab6ad63c-760b-4f74-899b-d4ea4473fc0d)

**Comment by ocornut** [comment]
For the win32 backend, try setting io. ConfigViewportsNoDefaultParent = true;
Unfortunately neither SDL or GLFW allow manipulating platform-level parent-child relationships.

(Please try to read the Glossary to use our standard terms in your viewport-related issues)

**Comment by dgm3333** [comment]
Don't you ever sleep?!
ConfigViewportsNoDefaultParent means the entire main viewport is then brought over top of any other viewports so isn't what I was after.

I'm at risk of asking an XY problem/question so background here:
I want the flexibility of a main viewport which can be used for docking a number of children on a secondary screen - as this is convenient for keeping the bulk of information in one place. However I also want to be able to scatter children over other applications to display additional information (eg point to locations and information that is being monitored - this is for an automation module for managing multiple pieces of medical software - so users need to be easily able to check that the locations my app is monitoring or controlling are accurate, or to put a child displaying information right where it is needed in another app.
Additionally I would ideally like to use the location/dimensions of children to allow users to set their own click zones/etc (those children could then be hidden again when not needed)
I also don't want the wasted title decoration (I want a Visual Studio-like style).
Thus I want the separate viewports to stay on top , but if I create a main viewport with no decoration it can't be dragged without additional coding of client clicking behaviour (which ImGui has already done as viewports), so this isn't ideal either - although this has the advantage that it's always behind the other viewports so would otherwise be my solution. 

So...
Am I correct in thinking that (assuming the other bugs with dragging across monitors were resolved) that if I created a main viewport, and immediately hid that then I could then create a child (which is then draggable) to use as the primary dock. Then all I have to do is lock that at the bottom of the z-order...

**Comment by ocornut** [comment]
Please do pay attention to the Glossary (particularly https://github.com/ocornut/imgui/wiki/Glossary#multi-viewports-terms)

It looks like you are using terms such as "child" for not what they are here. Considering the complexity of multi-viewport related issues + the general amount of issues being overwhelming and frequently delayed, if your post is hard to parse and ambiguous it is likely to make issue tackled years later.

**Comment by dgm3333** [comment]
Apologies. I have edited to (hopefully) align with the glossary and clarify some of the complexity (and a second apology as it's also off topic so I've broken another rule - perhaps I should start a new thread?).
